### PR TITLE
Inform mod-permissions when module is disabled OKAPI-982

### PR DESCRIPTION
**Pull request by adamdickmeiss**
1. Removing module sends module with empty list via _tenantPermission so that permissions module may mark those permissions as no longer in use by the module.
2. Config option: `enable_system_auth` to control whether to call auth even if it's enabled. Highly insecure, so this is something that can not be controlled at runtime.
3. install parameter `invoke` is a string rather than a `boolean`. Values `true` and `false` honored as before. Other value is regular expression for controlling which modules to invoke.

### CODE REVIEW

File: okapi-core/src/main/java/org/folio/okapi/util/TenantInstallOptions.java [hunk 1]
```diff
@@ -69,12 +69,23 @@ public boolean getDepCheck() {
...
 
-  public boolean getInvoke() {
-    return invoke;
+  /**
+   * Check if module is to be invoked during install.
+   * @param id module ID.
+   * @return whether to invoke the module.
+   */
+  public boolean checkInvoke(String id) {
+    if (invoke == null || "true".equals(invoke)) {
```
Comment (craigmcnally) [inline okapi-core/src/main/java/org/folio/okapi/util/TenantInstallOptions.java:82]: Do we want to try and handle "True" / "TRUE" / etc. as well?  `Boolean.parseBoolean(invoke)` might be a better check here?

File: okapi-core/src/main/java/org/folio/okapi/util/TenantInstallOptions.java [hunk 2]
```diff
@@ -69,12 +69,23 @@ public boolean getDepCheck() {
...
+      return true;
+    }
+    if ("false".equals(invoke)) {
```
Comment (craigmcnally) [inline okapi-core/src/main/java/org/folio/okapi/util/TenantInstallOptions.java:85]: Likewise, do we want to try and handle False / FALSE /etc. here?  maybe via something like `"false".equalsIgnoreCase(invoke)`

File: okapi-core/src/main/java/org/folio/okapi/util/TenantInstallOptions.java (hunk 1 above)
Comment (craigmcnally) [reply okapi-core/src/main/java/org/folio/okapi/util/TenantInstallOptions.java:82]: or as suggested below use `String.equalsIgnoreCase(...)` ?

File: okapi-core/src/main/java/org/folio/okapi/managers/TenantManager.java [hunk 3]
```diff
@@ -651,31 +650,41 @@ private void fireTimer(Tenant tenant, ModuleDescriptor md, RoutingEntry re, Stri
...
+          pl = new PermissionList(mdFrom.getId(), new Permission[0]);
+        }
         break;
       default:
         return Future.failedFuture(new OkapiError(ErrorType.USER,
             "Unknown version of _tenantPermissions interface in use " + permIntVer + "."));
     }
     String pljson = Json.encodePrettily(pl);
-    logger.debug("tenantPerms Req: {}", pljson);
+    logger.info("tenantPerms Req: {}", pljson);
```
Comment (craigmcnally) [inline okapi-core/src/main/java/org/folio/okapi/managers/TenantManager.java:687]:
```suggestion
    logger.info("_tenantPermissions Req: {}", pljson);
```

**Review by craigmcnally — APPROVED** [review]
Looks good - just a few minor suggestions - You can take them or leave them.

File: okapi-core/src/main/java/org/folio/okapi/util/TenantInstallOptions.java (hunk 1 above)
Comment (adamdickmeiss) [reply okapi-core/src/main/java/org/folio/okapi/util/TenantInstallOptions.java:82]: The code elsewhere is pretty strict with this .. In ModuleUtil.getParmBoolean where anything but `true` or `false` will be reported as error. So I think we should make it strict to keep the style.

File: okapi-core/src/main/java/org/folio/okapi/util/TenantInstallOptions.java (hunk 2 above)
Comment (adamdickmeiss) [reply okapi-core/src/main/java/org/folio/okapi/util/TenantInstallOptions.java:85]: :-)

File: okapi-core/src/main/java/org/folio/okapi/managers/TenantManager.java (hunk 3 above)
Comment (adamdickmeiss) [reply okapi-core/src/main/java/org/folio/okapi/managers/TenantManager.java:687]: fine

File: okapi-core/src/main/java/org/folio/okapi/util/TenantInstallOptions.java (hunk 1 above)
Comment (adamdickmeiss) [reply okapi-core/src/main/java/org/folio/okapi/util/TenantInstallOptions.java:82]: Someday, someone will make a `true`module.

**Pull request merged**